### PR TITLE
[vcpkg baseline][libao] Disable dlfcn check under windows

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**
In an internal CI test, libao installation failed with following errors:
```
.././../src/1.2.2-d87489a217.clean/src/audio_out.c(291): error C2065: 'DIR': undeclared identifier
.././../src/1.2.2-d87489a217.clean/src/audio_out.c(291): error C2065: 'plugindir': undeclared identifier
.././../src/1.2.2-d87489a217.clean/src/audio_out.c(301): error C2037: left of 'd_name' specifies undefined struct/union 'dirent'
.././../src/1.2.2-d87489a217.clean/src/audio_out.c(301): error C2198: 'strlen': too few arguments for call
.././../src/1.2.2-d87489a217.clean/src/audio_out.c(301): error C2057: expected constant expression
.././../src/1.2.2-d87489a217.clean/src/audio_out.c(301): error C2466: cannot allocate an array of constant size 0
```
This issue only occurs when dlfcn-win32 installed before libao in Windows. Libao always check dlfcn.h and enable the macro HAVE_DLOPEN if dlfcn.h is exists, then the DIR struct will be compiled, in windows environment this will cause the above errors to appear.

For fixing this issue, I passed a cache variable to disable the search of dlfcn under windows

No feature need to be tested.

### CODE REVIEW

**Comment by dg0yt** [comment]
Does it need a patch, or could you just pass a corresponding cache vale (`ac_cv_...=no` or similar) to `configure`? (See `config.log...` for `ac_cv_...` variables resulting from detection.)

**Comment by Cheney-W** [comment]
@dg0yt Thanks for your guidance!

File: ports/libao/portfile.cmake [hunk 1]
```diff
@@ -1,5 +1,6 @@
 if(VCPKG_TARGET_IS_WINDOWS)
```
Comment (JackBoosY) [inline ports/libao/portfile.cmake:1]:
```suggestion
set(NO_DLFCN )
if(VCPKG_TARGET_IS_WINDOWS)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/libao/portfile.cmake (hunk 1 above)
Comment (Cheney-W) [reply ports/libao/portfile.cmake:2]: Thanks for your remider!

**Review by JackBoosY — APPROVED** [review]
LGTM if the tests passed.

**Comment by JackBoosY** [comment]
@vicroms Please consider to merge this PR first.

File: ports/libao/portfile.cmake [hunk 2]
```diff
@@ -1,5 +1,7 @@
+set(NO_DLFCN )
```
Comment (dg0yt) [inline ports/libao/portfile.cmake:1]: This bad CMake style. It unset the variable in the current scope, unhiding a possible value from a parent scope. For a clean variable, always use:
~~~cmake
set(VAR "")
~~~

Comment (dg0yt) [reply ports/libao/portfile.cmake:1]: https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/cmake-guidelines.md

**Pull request merged**